### ISSUE TITLE
Application Style gets applied eventhough separate style provided

### ISSUE THREAD

**Issue by PaulAndersonS** [dotnet/maui#8439]
### Description

I have a custom view of type ContentView. In that i have a HorizontalStackLayout with two labels. I have set a new style for HorizontalStackLayout as below

`HorizontalStackLayout hStackLayout = new HorizontalStackLayout()
            {
                Style = new Style(typeof(HorizontalStackLayout)),
                HorizontalOptions = LayoutOptions.Start,
                VerticalOptions = LayoutOptions.Fill,
            };`

Also tried setting Style for the CustomView using resources like below 

`<local:CustomView>
                <local:CustomView.Resources>
                    <Style TargetType="HorizontalStackLayout">
                        <Setter Property="Spacing" Value="50"/>
                    </Style>
                </local:CustomView.Resources>
            </local:CustomView>`

But both of the approach not getting applied and the Style present in the Styles.xaml in the Application Resources gets applied.

Facing this issue in Android and Windows. Not yet tried in iOS and MacCatalsyt.

Sample:
[MauiApp15.zip](https://github.com/dotnet/maui/files/9019357/MauiApp15.zip)
 

### Steps to Reproduce

Run the attached sample.

### Version with bug

6.0.400 (current)

### Last version that worked well

6.0 Release Candidate 2 or older

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Affected platform versions

ANDROID 12

**Comment by janseris** [comment]
@rachelkang 

>Would you please reshare the sample in the form of a GitHub repository so that we may investigate further? We unfortunately are only able to accept samples in that form.

That is not true what you write about github repositories being the only accepted source. That was not needed until I saw you writing this some days ago. 
Also, imagine being a developer having 50 random repos for every maui issue out there if I did like you wrote. That sounds really silly.

**Comment by hartez** [comment]
> Also, imagine being a developer having 50 random repos for every maui issue out there if I did like you wrote. That sounds really silly.

You could use one repo and multiple branches.

**Comment by kristinx0211** [comment]
verified repro on windows and android with above project.

**Comment by jfversluis** [comment]
@janseris in the bug template we already ask for reproductions through GitHub and [under the link](https://github.com/dotnet/maui/blob/main/.github/repro.md) there is more information about why a reproduction project is imporant and we repeat the request for providing it through a GitHub repo

![image](https://user-images.githubusercontent.com/939291/176866735-8b4b1cb7-fa8f-48da-b36b-fab7a45d3994.png)

I'm sorry if that is not something you like, but something that we are going to enforce from a security standpoint. I will add a bit of clarification in the link above on why that is. Thanks for your cooperation!

**Comment by mattleibow** [comment]
Looking at this code for future things and there are 3 "styles" in this sample:
   1. App style
   2. Main page style 
   3. Blank C# style

It appears that the new blank style set in C# is preventing the main page style from applying - but not the app style.

**Comment by homeyf** [comment]
Verified this issue with Visual Studio Enterprise 17.7.0 Preview 1.0. Can repro on Windows and Android Platform with sample project.
[MauiApp15.zip](https://github.com/dotnet/maui/files/9019357/MauiApp15.zip)
![image](https://github.com/dotnet/maui/assets/82450723/a05f3f10-f45c-4c98-a71b-28b01513d806)
